src/Xamarin.Android.Build.Tasks/Tasks/NdkUtils.cs

```diff
@@ -16,19 +17,28 @@ namespace Xamarin.Android.Tasks
 {
 	public static class NdkUtil
 	{
+		// We need it to work fine during our tests which are executed at the same time in various threads.
+		[ThreadStatic]
 		static bool usingClangNDK;
 
 		public static bool UsingClangNDK => usingClangNDK;
 
+		public static bool Init (string ndkPath)
+		{
+			return Init (null, ndkPath); // For tests which don't have access to a TaskLoggingHelper
+		}
+
 		public static bool Init (TaskLoggingHelper log, string ndkPath)
 		{
 			Version ndkVersion;
 			bool hasNdkVersion = GetNdkToolchainRelease (ndkPath ?? "", out ndkVersion);
 
 			if (!hasNdkVersion) {
-				log.LogCodedError ("XA5101",
```

Comment:
Perhaps we could just do `log?.LogCodedError (...)`, which results in only a 1 line change?